### DISCUSSION TITLE
Help with API

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [directus/directus#6755] [Archive]
Hello, I have three questions ..

1. If I make field groups, can I set that the fields are also grouped in the Json API? Or can I do it differently somehow?

Example: I have a collection with cars and these cars have the group specifications, now I want that in the json api the whole thing is grouped under "specifications"
\
\
2. Can I somehow have several collections given to me with one get request? Or I can create a data model in which I can combine several collections and then display it.
\
\
3. Can I specify what is attached to a standard input?

Example: I have the collection with the cars again and I want to specify the width in the specifications, but I prefer to just specify a number and then automatically add "m" for meters. Or I have an asset and want to automatically append "directus.domain.de/assets/" to it in the API.
\
\
I hope someone can help me! :)

**Comment by Lucas-Gruber** [comment]
ps: i'm sorry with "group in the api" i actually meant i think a .. nested array or something ... sorry i'm pretty new to json 😅

**Comment by rijkvanzanten** [comment]
> If I make field groups, can I set that the fields are also grouped in the Json API? Or can I do it differently somehow?

No, the only nested field groups are through a raw JSON field, or a relational field at this point.

> 2. Can I somehow have several collections given to me with one get request? Or I can create a data model in which I can combine several collections and then display it.

Yes, you can use relationships to connect different datasets, and use the `fields` parameter / GraphQL to fetch any specific structure you need

> 3. Can I specify what is attached to a standard input?

You could use [a hook](https://docs.directus.io/guides/api-hooks/#custom-api-hooks) to dynamically alter the IO 👍🏻

**Reply by Lucas-Gruber** [reply]
Okay thank you! But when i use Relationships, i don't get the Content of Fields. How can i do this?

edit: i mean i only get the ID but i need also the other fields 😅

**Reply by philleepflorence** [reply]
Hi, hope this helps answer your question on relationships and fields:

- [The Directus Application - Concept Introduction](https://docs.directus.io/concepts/application/)
- [Fields: Query Parameters - API Reference](https://docs.directus.io/reference/api/query/#fields)
    - Choose the fields that are returned in the current dataset.
    - This parameter supports dot notation to request nested relational fields.

**Reply by Lucas-Gruber** [reply]
Thank you, but i don't understand the M2A fields-parameter. Can you maybe help me? 😅

**Reply by philleepflorence** [reply]
Hopefully the [Directus Developer Community on Discord](https://docs.directus.io/getting-started/introduction/#community) could offer more assistance...

> NOTE: 
> Many-to-Any is one of the different relationships - see [Relationships - Concept Reference](https://docs.directus.io/concepts/relationships/)
> The [examples](https://docs.directus.io/reference/api/query/#examples) should cover other common relationships.

Or, if you have any specific questions regarding: *M2A fields-parameter*?

**Reply by Lucas-Gruber** [reply]
Okay thanks, I have a specific question yes. I want to merge different collections (or rather certain fields) that can be done with M2A Relations according to docs. For example, I now have the "car_description" field from the "cars" collection and the "company_countrys" field from the "companys" collection and want to display these fields. but I don't know exactly in which collection I should create the relation, and with which parameter I can then output it.

**Reply by rijkvanzanten** [reply]
@Thomas-Blakeney That sounds like a regular many-to-one approach. A "car" will have a "manufacturer" (which would be a m2o to "companies"), after which you can pull in all the info from the "companies" record under the "manufacturer" field in the API 👍🏻

**Reply by Lucas-Gruber** [reply]
oh sorry i forgot to mention that i have a separate page for every car and for every company (unfortunately i don't know an example anymore but i have more collections) and i want to display them all together with certain fields.

**Reply by rijkvanzanten** [reply]
Yeah, that doesn't change anything 🙂

**Reply by Lucas-Gruber** [reply]
okay thank you! :smile: And with what argument can i now get the fields?

**Reply by rijkvanzanten** [reply]
Yeah, so TL;DR:

* You setup collections for all "entities" that you have (eg "cars" and "companies")
* Add relationships that make sense in the context of your project (eg each "car" has a "company" that made it)
* You can now fetch the fields (relationally) through the `/items/cars` endpoint, with something like `?fields=id,name,company.name`

**Reply by Lucas-Gruber** [reply]
okay thanks! Now there is only one thing left, what if, for example, I have several cars and want to list them?

**Reply by rijkvanzanten** [reply]
That's the example I just shared 👍🏻 See https://docs.directus.io/reference/api/items/ for more information

**Reply by Lucas-Gruber** [reply]
Sorry but i don't understand what you are meaning.

**Reply by Lucas-Gruber** [reply]
which relation i need to choose?

**Reply by Lucas-Gruber** [reply]
ps: i have collections with multiple items and collections that are treat as single object

**Reply by rijkvanzanten** [reply]
Have you seen https://docs.directus.io/concepts/relationships/ ? That describes the different relationship types

**Reply by Lucas-Gruber** [reply]
Yes, But no relationtype Working, or i make everything wrong.
